### ISSUE TITLE
[preset: firebase] [WITH FIX] Hosting deployment error "The system cannot find the path specified."

### ISSUE THREAD

**Issue by davidstackio** [unjs/nitro#615]
### Environment

Trying to deploy a Nuxt 3 RC 12 (`v3.0.0-rc.12`) app to firebase hosting on Windows.

Related discussions and issues:
https://github.com/nuxt/nuxt.js/issues/15185
https://github.com/nuxt/nuxt.js/issues/14439
https://github.com/unjs/nitro/issues/236
https://github.com/nuxt/framework/discussions/1873
https://github.com/nuxt/framework/discussions/6608

### Reproduction

Attempt to deploy a Nuxt 3 app to Firebase following these instructions:
https://nitro.unjs.io/deploy/providers/firebase

### Describe the bug
Got a `"The system cannot find the path specified."` error when I tried to deploy.

**This is what I did to fix it...**

Added a [`predeploy`](https://firebase.google.com/docs/hosting/test-preview-deploy#hooks) script to my `firebase.json`:

```
"hosting": {
    "predeploy": [
      "npm run build && cd .output/server && npm install"
    ],
    ...
```

This requires updating `nuxt.config` (so we don't need the `NITRO_PRESET=firebase` bit) to:
```
export default defineNuxtConfig({
  nitro: {
    preset: 'firebase'
  }
})
```

### Logs

```sh
Partial output log. Key line is `The system cannot find the path specified.`

...
Σ Total size: 15.7 MB (3.07 MB gzip)
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
i  artifactregistry: ensuring required API artifactregistry.googleapis.com is enabled...
+  functions: required API cloudbuild.googleapis.com is enabled
+  functions: required API cloudfunctions.googleapis.com is enabled
+  artifactregistry: required API artifactregistry.googleapis.com is enabled
i  functions: preparing codebase default for deployment
The system cannot find the path specified.

Error: An unexpected error has occurred.
```

**Comment by davidstackio** [comment]
On a related note, we don't need the `"site": "<your_project_id>",` in `firebase.json` as [it is no longer required](https://firebase.google.com/docs/hosting/full-config#firebase-json_example). It is specified at runtime or in the `.firebaserc` file like this:

```
{
  "projects": {
    "default": "<your_project_id>"
  }
}
```

**Comment by dosstx** [comment]
> "predeploy": [
>       "npm run build && cd .output/server && npm install"
>     ],

If I do that, I then get the following error:

`Error: Assertion failed: resolving hosting target of a site with no site name or target name. This should have caused an error earlier`

So, I put it back in.

**Comment by davidstackio** [comment]
In recent versions of Nuxt and Firebase using the `firebase deploy` command no longer works. I got an error when deploying the function about package-lock having mismatched versions. I spent a few hours trying to figure out what was wrong until I came up with the below solution.

Now I use `npm run deploy` to deploy. Excerpt from my `package.json`:
```
"scripts": {
    "build": "nuxt build",
    "deploy": "npm run build && npx firebase deploy",
```
